File: back-end/src/main/java/CRM4RTONOCTURNOA/CRM/entity/Cotizacion.java
package CRM4RTONOCTURNOA.CRM.entity;




import java.sql.Timestamp;
import java.util.HashSet;
import java.util.Set;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.MappedCollection;
import org.springframework.data.relational.core.mapping.Table;

import lombok.Data;



@Data
@Table("reporte\".\"cotizaciones")
public class Cotizacion {

    @Id
    @Column("cotizacion_id")
    private long cotizacionId;

    private String nro;

    private Timestamp fecha;

    @Column("cliente_id")
    private long clienteId;

    private String observacion;

    @MappedCollection(idColumn = "cotizacion_id")
    Set<DetalleCotizacion> detalleCotizacion = new HashSet<>();

    
}

File: fron-end/src/app/reporte/cotizacion/cotizacion.ts
import { DetalleCotizacion } from "./detalleCotizacion";

export class Cotizacion{

  constructor(
    public cotizacionId: number,
    public nro: string,
    public fecha: Date,
    public clienteId: number,
    public detalleCotizacion: DetalleCotizacion[]

  )
  {}

}
